sensor/common/networkflow/manager/manager_impl.go

```diff
@@ -321,7 +336,14 @@ func (m *networkFlowManager) enrichConnections(tickerC <-chan time.Time) {
 	}
 }
 
+func (m *networkFlowManager) getCurrentContext() context.Context {
+	m.ctxMutex.Lock()
+	defer m.ctxMutex.Unlock()
+	return m.pipelineCtx
+}
```

Comment:
Does this make sense if we only return the context? This is not a reading operation per se...
The caller of this func will be operating on the context anyway, so the mutex needs to be locked additionally outside of this func.

Having said that, I believe that we can skip locking the mutex here.

Reply:
I ran the tests -race without this lock and they pass. WDYT?

Reply:
Odd. I think it should race. Because it's being overwritten in `resetContext`. I think the problem is that the tests are not simulating a real environment, where `Notify` can be called at the same time as the ticker loop. In the tests, they are two synchronous calls: see [here](https://github.com/stackrox/stackrox/pull/7031/files#diff-bdc111f86b91adbfb88275e12b1de7bda68012b3a6c13e4c84c4929f4b0fb445R684-R685).

If you pass a real ticker to the tests I'm reasonably confident it will give you a data race.

Reply:
Okay, let's keep it then